msg/CollisionObject.msg

```diff
@@ -1,27 +1,36 @@
-# a header, used for interpreting the poses
+# A header, used for interpreting the poses
 Header header
 
-# the id of the object (name used in MoveIt)
+# The id of the object (name used in MoveIt)
 string id
 
 # The object type in a database of known objects
 object_recognition_msgs/ObjectType type
 
-# the the collision geometries associated with the object;
-# their poses are with respect to the specified header
+# The collision geometries associated with the object.
+# Their poses are with respect to the specified header
 
-# solid geometric primitives
+# Solid geometric primitives
 shape_msgs/SolidPrimitive[] primitives
 geometry_msgs/Pose[] primitive_poses
 
-# meshes
+# Meshes
 shape_msgs/Mesh[] meshes
 geometry_msgs/Pose[] mesh_poses
 
-# bounding planes (equation is specified, but the plane can be oriented using an additional pose)
+# Bounding planes (equation is specified, but the plane can be oriented using an additional pose)
 shape_msgs/Plane[] planes
 geometry_msgs/Pose[] plane_poses
 
+# Named frames on the object. Use these to define points of interest on the object that you want
+# to plan with (e.g. screwdriver_tip, kettle_spout, mug_base).
+# The length of these vectors has to be identical.
+# The names of named frames in the scene need to be unique! If both "screwdriver" and "pencil" have
+# a named framed called "tip", it will cause silent errors! Use "screwdriver_tip" and "pencil_tip" instead,
+# or submit a pull request to fix this limitation.
+string[] frame_names
```

Comment:
Maybe use `subframe_ids` and `subframe_poses`.

Reply:
I'm not opposed, but we should probably discuss in [#1439](https://github.com/ros-planning/moveit/pull/1439).

Reply:
Changed as suggested